corehq/apps/hqmedia/tasks.py

```diff
@@ -185,7 +185,8 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
                     if extension not in MULTIMEDIA_EXTENSIONS:
                         file_cache[path] = data
 
-        errors.extend(find_missing_locale_ids_in_ccz(file_cache))
+        if toggles.LOCALE_ID_INTEGRITY.enabled(app.domain):
```

Comment:
Did this check pass for the AWW app, but not for the IT app? Would it make sense to make this an add on so it could be enabled app by app?

Reply:
It fails for AWW and IT, passes for LS. It shouldn't be an add-on because it isn't meant to be a user-facing setting, it really ought to be incorporated into the product (I'll follow up with solutions about other domains using case tiles). Also, I'd rather force ICDS to update all their apps so we get more info about locale id bugs. Will reconsider if it turns out that it's difficult for them to update all apps because streams/performance mean no app change is trivial.

Reply:
i believe add ons are stored on the app doc?
If thats true we would be leaving 'Locale_id_integrity' : True/False on app docs which would not mean anything in future once this is not needed anymore and removed.
Hence, i'd push to keep in as a FF.

Reply:
I'm not planning to remove this in the future - the flag, maybe, but not the feature. The need for app strings to contain everything in the suite, when those two files are generated by very different code paths, is going to be an ongoing area of risk.

Reply:
i meant the ff. the feature is great.

Reply:
@mkangia oops, I totally misread your comment.